src/app/components/Navbar.tsx

```diff
@@ -93,10 +122,41 @@ const Navbar = () => {
   );
 };
 
-export default function NavbarComponent() {
+export default Navbar;
+
+const NavHeading = () => {
+  const pathname = usePathname();
+
   return (
-    <Suspense>
-      <Navbar />
-    </Suspense>
+    <div className="flex items-center gap-1">
+      {pathname !== '/' && <BackButton className="text-white" />}
+      <Link href="/">
+        <h5 id="drawer-navigation-label" className="font-bold text-gray-300 uppercase dark:text-gray-400">
+          Dua Collection
+        </h5>
+      </Link>
+    </div>
   );
-}
+};
+
+const MenuLinks = ({ toggle }: INavComponentProps) => {
+  const searchParams = useSearchParams();
+
+  return (
+    <Fragment>
+      {menu.map(({ name, url }, index) => {
+        const isActive = searchParams.get('tag') === url.query?.tag;
```

Comment:
This kind of small component should be stateless and provide the props from the parent component. Keeping the components stateless as much as possible help to tinker with state and side affects easily.